sql/core/src/main/scala/org/apache/spark/sql/sources/interfaces.scala

```diff
@@ -130,8 +130,9 @@ trait SchemaRelationProvider {
 trait StreamSourceProvider {
   def createSource(
       sqlContext: SQLContext,
-      parameters: Map[String, String],
-      schema: Option[StructType]): Source
+      partitionColumns: Option[StructType],
+      dataSchema: Option[StructType],
```

Comment:
nit: could we order these the same as below: dataSchema, partitionColumns, parameters